auth/endpoint_id_extractor.go

```diff
@@ -7,81 +7,47 @@ import (
-	EndpointIDExtractorTypeHeader EndpointIDExtractorType = "header"
-)
-
-// IsValid ensure the endpoint ID extractor type is supported.
-func (e EndpointIDExtractorType) IsValid() bool {
-	switch e {
-	case EndpointIDExtractorTypeURLPath, EndpointIDExtractorTypeHeader:
-		return true
-	default:
-		return false
+// extractEndpointID extracts the endpoint ID from the HTTP request.
+// It first attempts to extract the ID from the header, and if that fails, it falls back to the URL path.
+// If both extraction methods fail, it returns an error.
+func extractEndpointID(req *envoy_auth.AttributeContext_HttpRequest) (string, error) {
+	if id := extractFromHeader(req); id != "" {
+		return id, nil
 	}
-}
-
-// EndpointIDExtractor defines an interface for extracting an endpoint ID from a given source.
-// This could be a URL path, HTTP header, etc...
-type EndpointIDExtractor interface {
-	// extractGatewayEndpointID extracts the endpoint ID from the check request.
-	extractGatewayEndpointID(req *envoy_auth.AttributeContext_HttpRequest) (string, error)
-}
-
-// URLPathExtractor satisfies the EndpointIDExtractor interface.
-var _ EndpointIDExtractor = &URLPathExtractor{}
-
-// URLPathExtractor is an implementation of the EndpointIDExtractor interface
-// that extracts the endpoint ID from the URL path.
-type URLPathExtractor struct{}
-
-// extractGatewayEndpointID extracts the endpoint ID from the URL path.
-// The endpoint ID is expected to be the first segment of the path after the pathPrefix (/v1/)
-//
-// eg. http://eth.path.grove.city/v1/1a2b3c4d -> endpointID = "1a2b3c4d"
-func (p *URLPathExtractor) extractGatewayEndpointID(req *envoy_auth.AttributeContext_HttpRequest) (string, error) {
-	path := req.GetPath()
-
-	if strings.HasPrefix(path, pathPrefix) {
-		segments := strings.Split(strings.TrimPrefix(path, pathPrefix), "/")
-		if len(segments) > 0 && segments[0] != "" {
-			return segments[0], nil
-		}
+	if id := extractFromPath(req); id != "" {
+		return id, nil
 	}
-
-	return "", fmt.Errorf("endpoint-id not provided in the URL path")
+	return "", fmt.Errorf("endpoint ID not provided in header or path")
 }
 
-// HeaderExtractor satisfies the EndpointIDExtractor interface.
-var _ EndpointIDExtractor = &HeaderExtractor{}
-
-// HeaderExtractor is an implementation of the EndpointIDExtractor interface
-// that extracts the endpoint ID from the HTTP headers.
-type HeaderExtractor struct{}
-
-// ExtractGatewayEndpointID extracts the endpoint ID from the HTTP headers.
-// The endpoint ID is expected to be in the "endpoint-id" header.
-//
-// eg. Header = "endpoint-id: 1a2b3c4d" -> endpointID = "1a2b3c4d"
-func (h *HeaderExtractor) extractGatewayEndpointID(req *envoy_auth.AttributeContext_HttpRequest) (string, error) {
+// extractFromHeader extracts the endpoint ID from the headers.
+// It returns the endpoint ID if found and non-empty, otherwise an empty string.
+// Example: Header = "endpoint-id: 1a2b3c4d" -> endpointID = "1a2b3c4d"
```

Comment:
Isn't the best practice for it to be `Endpoint-Id` instead of `endpoint-id`?

Reply:
In practice it doesn't matter but yes you're right it should be, I will fix the casing.